File: subly_android/app/src/main/java/net/cynreub/subly/domain/model/Category.kt
package net.cynreub.subly.domain.model

import java.util.UUID

data class Category(
    val id: UUID,
    val name: String,        // internal identifier, e.g. "STREAMING"
    val displayName: String, // user-facing, e.g. "Streaming"
    val emoji: String,
    val colorHex: String
) {
    companion object {
        // Fixed UUIDs matching AppDatabase migration constants — never change these
        val ID_STREAMING  = UUID.fromString("00000000-0000-4000-a000-000000000001")
        val ID_MAGAZINE   = UUID.fromString("00000000-0000-4000-a000-000000000002")
        val ID_SERVICE    = UUID.fromString("00000000-0000-4000-a000-000000000003")
        val ID_MEMBERSHIP = UUID.fromString("00000000-0000-4000-a000-000000000004")
        val ID_CLUB       = UUID.fromString("00000000-0000-4000-a000-000000000005")
        val ID_UTILITY    = UUID.fromString("00000000-0000-4000-a000-000000000006")
        val ID_SOFTWARE   = UUID.fromString("00000000-0000-4000-a000-000000000007")
        val ID_OTHER      = UUID.fromString("00000000-0000-4000-a000-000000000008")

        val DEFAULTS = listOf(
            Category(ID_STREAMING,  "STREAMING",  "Streaming",  "📺", "#E91E63"),
            Category(ID_MAGAZINE,   "MAGAZINE",   "Magazine",   "📰", "#9C27B0"),
            Category(ID_SERVICE,    "SERVICE",    "Service",    "⚙️", "#2196F3"),
            Category(ID_MEMBERSHIP, "MEMBERSHIP", "Membership", "🏷️", "#FF9800"),
            Category(ID_CLUB,       "CLUB",       "Club",       "🎯", "#4CAF50"),
            Category(ID_UTILITY,    "UTILITY",    "Utility",    "💡", "#607D8B"),
            Category(ID_SOFTWARE,   "SOFTWARE",   "Software",   "💻", "#00BCD4"),
            Category(ID_OTHER,      "OTHER",      "Other",      "📦", "#795548")
        )

        val DEFAULT_ID = ID_OTHER
    }
}

File: subly_ios/Subly/Subly/Domain/Models/Category.swift
import Foundation

struct Category: Identifiable, Equatable, Codable {
    var id: UUID
    var name: String        // internal key, e.g. "STREAMING"
    var displayName: String // user-facing, e.g. "Streaming"
    var emoji: String
    var colorHex: String

    // Fixed UUIDs matching Android AppDatabase constants — never change these
    static let idStreaming  = UUID(uuidString: "00000000-0000-4000-a000-000000000001")!
    static let idMagazine   = UUID(uuidString: "00000000-0000-4000-a000-000000000002")!
    static let idService    = UUID(uuidString: "00000000-0000-4000-a000-000000000003")!
    static let idMembership = UUID(uuidString: "00000000-0000-4000-a000-000000000004")!
    static let idClub       = UUID(uuidString: "00000000-0000-4000-a000-000000000005")!
    static let idUtility    = UUID(uuidString: "00000000-0000-4000-a000-000000000006")!
    static let idSoftware   = UUID(uuidString: "00000000-0000-4000-a000-000000000007")!
    static let idOther      = UUID(uuidString: "00000000-0000-4000-a000-000000000008")!
    static let defaultId    = idOther

    static let defaults: [Category] = [
        Category(id: idStreaming,  name: "STREAMING",  displayName: "Streaming",  emoji: "📺", colorHex: "#E91E63"),
        Category(id: idMagazine,   name: "MAGAZINE",   displayName: "Magazine",   emoji: "📰", colorHex: "#9C27B0"),
        Category(id: idService,    name: "SERVICE",    displayName: "Service",    emoji: "⚙️", colorHex: "#2196F3"),
        Category(id: idMembership, name: "MEMBERSHIP", displayName: "Membership", emoji: "🏷️", colorHex: "#FF9800"),
        Category(id: idClub,       name: "CLUB",       displayName: "Club",       emoji: "🎯", colorHex: "#4CAF50"),
        Category(id: idUtility,    name: "UTILITY",    displayName: "Utility",    emoji: "💡", colorHex: "#607D8B"),
        Category(id: idSoftware,   name: "SOFTWARE",   displayName: "Software",   emoji: "💻", colorHex: "#00BCD4"),
        Category(id: idOther,      name: "OTHER",      displayName: "Other",      emoji: "📦", colorHex: "#795548"),
    ]
}

struct CategoryWithCount: Identifiable {
    var category: Category
    var subscriptionCount: Int
    var id: UUID { category.id }
}
